src/operator/rnn-inl.h

```diff
@@ -140,14 +140,14 @@ inline int GetRnnParamSize(int num_layer,
       size *= 3;
       break;
   }
-  int size1 = (input_size + state_size + 2) * size;  // first layer size
-  int size2 = (state_size * direction + state_size + 2) * size;  // other layers size
+  index_t size1 = (input_size + state_size + 2) * size;  // first layer size
```

Comment:
Lets prefer size_t for sizes. Or do you think these values can be negative too ?

Reply:
Agreed - testing changes now.